zerver/tests/test_messages.py

```diff
@@ -2731,3 +2733,38 @@ def test_messages_for_ids(self) -> None:
 
         self.assertIn('class="user-mention"', new_message['content'])
         self.assertEqual(new_message['flags'], ['mentioned'])
+
+class MessageVisibilityTest(ZulipTestCase):
+    def test_get_first_visible_message_id_with_no_limit(self) -> None:
+        realm = get_realm("zulip")
+        realm.message_visibility_limit = None
+        realm.save()
+        cache_delete(realm_last_visible_message_id_cache_key(realm))
+
+        self.assertEqual(get_first_visible_message_id(realm), 0)
+
+    def test_get_first_visible_message_id_with_limit(self) -> None:
+        for i in range(5):
+            self.send_stream_message(self.example_email("othello"), "Scotland")
+
+        realm = get_realm("zulip")
+        num_messages = Message.objects.filter(sender__realm=realm).count()
```

Comment:
I would make this test deterministic by just zapping all the existing messages first, so do `Message.objects.delete()`.